### ISSUE TITLE
Incomplete memberdeclaration rule in Cpp grammar

### ISSUE THREAD

**Issue by romeo-cmd** [antlr/grammars-v4#1754]
I think the `explicitspecialization` rule is missing as an alternative within the rule

```
memberdeclaration

: attributespecifierseq? declspecifierseq? memberdeclaratorlist? ';'

| functiondefinition

| usingdeclaration

| static_assertdeclaration

| templatedeclaration

| aliasdeclaration

| emptydeclaration ;
```

I don't know if it was already possible in C++14 standard, but at least now I think it is possible to explicitly specialize (but not explicitly instanciate) a template within some class declaration. Therefore I would add the alternative `explicitspecialization` to the `memberdeclaration` parser rule.

**Comment by Marti2203** [comment]
Yes, this should be added. Will do now. I am making a big PR for the C++ grammar
